src/pages/api/events/discount-code.ts

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable no-console */
+// from https://wagmi.sh/examples/sign-in-with-ethereum
+
+/* eslint-disable consistent-return */
+import axios from 'axios';
+import { withIronSessionApiRoute } from 'iron-session/next';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { authUser } from 'utils/auth';
+import { sessionOptions, Session } from 'utils/session';
+
+const eventbriteAPI = process.env.NEXT_PRIVATE_EVENTBRITE_KEY;
+const orgID = process.env.NEXT_PRIVATE_EVENTBRITE_ORGID;
+
+const handler = async (
+  req: NextApiRequest,
+  res: NextApiResponse
+): Promise<void> => {
+  const { method } = req;
+  switch (method) {
+    case 'GET': {
+      const { eventCode } = req.query;
+      if (!eventCode) {
+        res.status(400).json({ error: 'expected eventCode in query params' });
+        break;
+      }
+      const { siwe } = req.session as Session;
+      const authRes = await authUser(siwe);
+      if (!authRes.valid) {
+        res.status(401).json({ error: authRes.errorMessage });
+      } else {
+        const code = `atx-${authRes.siwe.address}`;
```

Comment:
need to test this behavior but this could make it to where members can only use their discount code once.